exercises/queen-attack/queen-attack.js

```diff
@@ -0,0 +1,20 @@
+//
+// This is only a SKELETON file for the 'Queen Attack' exercise. It's been provided as a
+// convenience to get you started writing code faster.
+//
+
+export class QueenAttack {
+    constructor(params = { white: [], black: []}) {
+        this.white = params.white;
+        this.black = params.black;
```

Comment:
This is part of the implementation & we try to keep stub files free from this - so that users can choose different ways to implement. For example in this case, I may use something other than arrays and have `get` functions for white & black.

Can you please remove this and just have an empty constructor:

```js
constructor() {
  throw new Error("Remove this statement and implement this function");
}
```

Reply:
Alright. I was initially going to keep the constructor empty, but, in the test cases, the constructor expects an object with arrays for white and black, so I thought it would be nice to let the users know that.

Reply:
We understand the reasoning 😄 👍, it's just that with leaving it empty, students can choose to forward the object to a helper function, and are not _led_ by the initial naming etc.